processor/k8sprocessor/kube/client.go

```diff
@@ -71,6 +72,12 @@ func New(logger *zap.Logger, rules ExtractionRules, filters Filters, newClientSe
 	}
 	c.kc = kc
 
+	if newOwnerProviderFunc == nil {
+		newOwnerProviderFunc = newOwnerProvider
+	}
+
+	c.op = newOwnerProviderFunc(logger, kc, !limitsPodScope(filters))
```

Comment:
You might want to have this function able to return an error.  Given that it's caller-supplied, it's probably better to allow an error to be an additional return value.